### PR TITLE
Cobertura to use maven plugin rather than ant-run

### PR DESCRIPTION
**Pull request by duncangrant**
PLEASE DON'T MERGE YET
Seems to lose ability to output xml but does give us an aggregated report.

Also when run it seems to compile twice (presumably once without and then once with instrumentation) and you get warnings of the form:
WARNING: Duplicate name in Manifest: Manifest-Version.                                                                                  Ensure that the manifest does not have duplicate entries, and                                                                           
that blank lines separate individual sections in both your                                                                              
manifest and in the META-INF/MANIFEST.MF entry in the jar file.                                                                         

You seem to also get these warnings if you try to do an incremental build so I don't know if they're important.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -824,11 +823,11 @@
                     <artifactId>build-helper-maven-plugin</artifactId>
                     <version>1.7</version>
                 </plugin>
-                <plugin>
+                <!-- <plugin>
```
Comment (neykov) [inline pom.xml]: Remove if not used.

File: usage/rest-server/pom.xml [hunk 2]
```diff
@@ -246,7 +246,12 @@
             </resource>
         </resources>
         <plugins>
-
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <compilerId>javac</compilerId>
```
Comment (neykov) [inline usage/rest-server/pom.xml]: What's the reason for overriding the compiler here?

**Comment by neykov** [comment]
I tried running the tests with `mvn clean install -PCoverage` and get the following error:
```
[ERROR] Failed to execute goal org.codehaus.mojo:cobertura-maven-plugin:2.6:instrument (default) on project brooklyn-parent: Execution default of goal org.codehaus.mojo:cobertura-maven-plugin:2.6:instrument failed: Plugin org.codehaus.mojo:cobertura-maven-plugin:2.6 or one of its dependencies could not be resolved: Could not find artifact com.sun:tools:jar:0 at specified path /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/Contents/Home/../lib/tools.jar -> [Help 1]
```

**Comment by duncangrant** [comment]
What java version are you using?  I'm using oracle jdk 1.7 - Alasdair thinks it might be to do with this.

**Comment by neykov** [comment]
I am using Apple's Java 1.6.
Turns out that apple's build doesn't have tools.jar, it's merged with classes.jar. Cobertura depends on it so it fails.

Here's a related issue https://jira.codehaus.org/browse/MCOBERTURA-180.

Possible (temporary) solution: http://permalink.gmane.org/gmane.comp.java.maven-plugins.mojo.user/4320

**Comment by duncangrant** [comment]
The suggested fix is quite messy.  Perhaps it would be better to just accept that we can't run coverage on a mac running 1.6?

**Comment by neykov** [comment]
It's fine by me.

**Comment by ahgittin** [comment]
dropping java1.6 support coverage sounds fine

this runs fine even without #534 (fixed problem with antrun cobertura); it must be using a different directory structure.

however not only does this compile twice (which is understandable at least in `/src/main` as we need the instrumented and non-instrumented versions of classes), this **runs** all the tests twice which seems bad; acceptable but a major slow-down.

do we have a way to run the coverage check with both unit and integration tests activated?  the latter is important especially for packages such as `ssh`.

@duncangrant do you suggest we merge this now (with the faults which have been noted here) or close this and you reopen it when ready?

**Comment by ahgittin** [comment]
meant also to say:  it works well for me, on os x.  no `output.xml` as duncan g says.  i don't know if that matters (or indeed if the current antrun-based integration generates any such file).

**Comment by duncangrant** [comment]
The current ant-run based integration does produce output.xml.  This is useful for integration with jenkins and if we want to produce reports from the output.  For this reason I am going to close rather than merge until I can investigate the issue.

I hadn't notice it running the tests twice before.  According to @grkvlt this is because cobertura runs the tests once to confirm they pass and then instruments them and runs them again.

I haven't ever managed to get the integration tests to run on my machine so I still don't know what the coverage will look like for those.  Presumably it is possible to get both unit tests and integration.

**Comment by ahgittin** [comment]
if tests running twice is the normal thing then fine

would be nice to move to this "proper" build but no urgency, i support closing it, but no forgetting about it!

**Pull request closed without merging**